src/WinGetServer/WinGetServerManualActivation_Client.cpp

```diff
@@ -64,23 +61,112 @@ void InitializeRpcBinding()
+struct ServerProcessLauncher
 {
-    const std::filesystem::path& localAppDataPath = GetKnownFolderPath(FOLDERID_LocalAppData);
-    const std::filesystem::path& serverExePath = localAppDataPath / s_ServerExePath;
-    std::wstring commandLineInput = std::wstring{ serverExePath } + L" --manualActivation";
+    ServerProcessLauncher()
+    {
+        try
+        {
+            m_serverExePath = GetPackageLocation(s_ServerPackageFamilyName, s_ServerFileName) / s_ServerFileName;
+
+#ifndef USE_PROD_WINGET_SERVER
+            // The feature that allows directly launching a packaged process as long as it has a matching alias
+            // requires a failure to trigger, and the dev package is not ACL'd to force this to happen. Attempting
+            // to use the other code path results in an unpackaged server, causing other issues.
+            // We run the product code above to ensure that it is functioning properly, but then replace it with
+            // the path of the alias.
+            m_serverExePath = GetKnownFolderPath(FOLDERID_LocalAppData) / s_LocalAppDataRelativeServerExePath;
+#endif
+        }
+        catch (wil::ResultException& re)
+        {
+            m_hr = re.GetErrorCode();
+        }
+    }
+
+    HRESULT LaunchWinGetServerWithManualActivation()
+    {
+        RETURN_IF_FAILED(m_hr);
 
-    STARTUPINFO info = { sizeof(info) };
-    wil::unique_process_information process;
+        std::wstring commandLineInput = std::wstring{ m_serverExePath } + L" --manualActivation";
 
-    RETURN_LAST_ERROR_IF(!CreateProcessW(NULL, &commandLineInput[0], NULL, NULL, FALSE, 0, NULL, NULL, &info, &process));
+        STARTUPINFO info = { sizeof(info) };
+        wil::unique_process_information process;
 
-    // Wait for manual reset event from server before proceeding with COM activation.
-    wil::unique_event manualResetEvent = CreateOrOpenServerStartEvent();
-    manualResetEvent.wait(10000);
+        RETURN_LAST_ERROR_IF(!CreateProcessW(NULL, &commandLineInput[0], NULL, NULL, FALSE, 0, NULL, NULL, &info, &process));
 
-    return S_OK;
-}
+        // Wait for manual reset event from server before proceeding with COM activation.
+        wil::unique_event manualResetEvent = CreateOrOpenServerStartEvent();
+        manualResetEvent.wait(10000);
+
+        return S_OK;
+    }
+
+private:
+    std::filesystem::path GetPackageLocation(std::wstring_view packageFamilyName, std::wstring_view fileName)
+    {
+        std::wstring pfn{ packageFamilyName };
+        UINT32 count = 0;
+        std::unique_ptr<PWSTR[]> fullNames;
+        UINT32 bufferLength = 0;
+        std::unique_ptr<WCHAR[]> buffer;
+        std::unique_ptr<UINT32[]> properties;
+
+        LONG result = FindPackagesByPackageFamily(pfn.c_str(), PACKAGE_FILTER_HEAD, &count, nullptr, &bufferLength, nullptr, nullptr);
+        THROW_WIN32_IF(result, result != ERROR_INSUFFICIENT_BUFFER);
+
+        for (size_t i = 0; i < 10 && result == ERROR_INSUFFICIENT_BUFFER; ++i)

```

Comment:
Is this for "in case a new package is added between our calls"?

Reply:
It is defensive, yes.